scripts/check_if_pr_is_low_risk.py

```diff
@@ -0,0 +1,292 @@
+                i += 1
+                if line.startswith('@@'):
+                    break
+            if i > 6 or i == len(file_diff_split):
+                python_utils.PRINT(
+                    'Skipped too many lines when parsing "%s" diff'
+                    % filename)
+                return [], {}
+            file_diffs[filename] = file_diff_split[i:]
+    return diff_files, file_diffs
+
+
+def lookup_pr(owner, repo, pull_number):
+    """Lookup a PR using the GitHub API.
+
+    Args:
+        owner: str. Owner of the repository the PR is in.
+        repo: str. Repository the PR is in.
+        pull_number: str. PR number.
+
+    Returns:
+        dict. JSON object returned by the GitHub API v3. This is an
+        empty dictionary if the response code from the GitHub API is not
+        200.
+    """
+    request = python_utils.url_request(
+        GITHUB_API_PR_ENDPOINT % (owner, repo, pull_number),
+        None,
+        {'Accept': 'application/vnd.github.v3+json'})
+    response = python_utils.url_open(request)
+    if response.getcode() != 200:
+        return {}
+    pr = json.load(response)
+    response.close()
+    return pr
+
+
+def check_if_pr_is_translation_pr(pr):
+    """Check if a PR is low-risk by virtue of being a translation PR.
+
+    To be a low-risk translation PR, a PR must:
+
+    * Be opened from a branch on the oppia/oppia repository.
+    * Be opened from the branch `translatewiki-prs`.
+    * Only change JSON files in `assets/i18n`.
+    * Not change the names of any files.
+
+    Args:
+        pr: dict. JSON object of PR from GitHub API.
+
+    Returns:
+        str. An empty string if the PR is a translation PR and low-risk,
+        else a message explaining why the PR is not low-risk.
+    """
+    source_repo = pr['head']['repo']['full_name']
+    if source_repo != 'oppia/oppia':
+        return 'Source repo is not oppia/oppia'
+    source_branch = pr['head']['ref']
+    if source_branch != 'translatewiki-prs':
+        return 'Source branch is not translatewiki-prs'
+    diff_files, _ = load_diff(pr['base']['ref'])
+    if not diff_files:
+        return 'Failed to load PR diff'
+    for old, new in diff_files:
+        if not old == new:
+            return 'File name change: %s -> %s' % (old, new)
+        if not re.match('^assets/i18n/[a-z-]+.json$', old):
+            return 'File %s changed and not low-risk' % old
+    return ''
+
+
+def _check_changelog_pr_diff(diff_files, file_diffs):
+    """Check whether a changelog PR diff is valid.
+
+    Args:
+        diff_files: list(tuple(str, str)). Changed files, each as a
+            tuple of (old name, new name).
+        file_diffs: dict(str, list(str)). Map from file names to the
+            lines of that file's diff.
+
+    Returns:
+        str. If the diff is not valid for a low-risk changelog PR, an
+        error message explaining why. Otherwise, an empty string.
+    """
+    for old, new in diff_files:
```

Comment:
instead of looping over all the files should we do this (not sure!):
if diff_files != ['AUTHORS', 'CONTRIBUTORS', 'CHANGELOG', 'core/templates/pages/about-page/about-page.constants.ts']:
    return '...'
package_diff = file_diffs[''package.json']

...

Reply:
Why would this help? We currently return early as soon as a risky change is detected, so for most PRs we won't actually loop through all the files. Therefore, I don't think your suggested change would make this any faster.